tests/orchestrator/test_sequential_orchestrator_with_retry.py

```diff
@@ -217,3 +252,137 @@ def test_failed_tokyo_hit_max_attempts():
+    add_hello_action(expected_state, ['Tokyo', 'Seattle', 'London'])
+    expected_state._output = ["Hello Tokyo!", "Hello Seattle!", "Hello London!"]
+    expected_state._is_done = True
+    expected = expected_state.to_json()
+
+    assert_valid_schema(result)
+    assert_orchestration_state_equals(expected, result)
+
+def test_concurrent_retriable_results_mixed_arrival():
+    failed_reason = 'Reasons'
+    failed_details = 'Stuff and Things'
+    context_builder = ContextBuilder('test_concurrent_retriable_unordered_results')
+    # one task succeeds, the other two fail at first, and succeed on retry
+    add_hello_failed_events(context_builder, 1, failed_reason, failed_details)
+    add_hello_completed_events(context_builder, 0, "\"Hello Tokyo!\"")
+    add_hello_failed_events(context_builder, 2, failed_reason, failed_details)
+    add_retry_timer_events(context_builder, 3)
+    add_retry_timer_events(context_builder, 4)
+    add_hello_completed_events(context_builder, 6, "\"Hello London!\"")
+    add_hello_completed_events(context_builder, 5, "\"Hello Seattle!\"")
+
+    result = get_orchestration_state_result(
+        context_builder, generator_function_concurrent_retries)
+
+    expected_state = base_expected_state()
+    add_hello_action(expected_state, ['Tokyo', 'Seattle', 'London'])
+    expected_state._output = ["Hello Tokyo!", "Hello Seattle!", "Hello London!"]
+    expected_state._is_done = True
+    expected = expected_state.to_json()
+
+    assert_valid_schema(result)
+    assert_orchestration_state_equals(expected, result)
+
+def test_concurrent_retriable_results_alternating_taskIDs():
+    failed_reason = 'Reasons'
+    failed_details = 'Stuff and Things'
+    context_builder = ContextBuilder('test_concurrent_retriable_unordered_results')
+
+    ## Schedule tasks
+
+    # one task succeeds, the other two fail at first, and succeed on retry
+    context_builder.add_task_scheduled_event(name='Hello', id_=0) # Tokyo task
+    context_builder.add_task_scheduled_event(name='Hello', id_=1) # Seattle task
+
+    context_builder.add_orchestrator_completed_event()
+    context_builder.add_orchestrator_started_event()
+
+    ## Task failures and timer-scheduling
+
+    # tasks fail "out of order"
+    context_builder.add_task_failed_event(
+        id_=1, reason=failed_reason, details=failed_details) # Seattle task
+    fire_at = context_builder.add_timer_created_event(2) # Seattle timer
+
+    context_builder.add_orchestrator_completed_event()
+    context_builder.add_orchestrator_started_event()
+
+    context_builder.add_task_failed_event(
+        id_=0, reason=failed_reason, details=failed_details) # Tokyo task
+    fire_at = context_builder.add_timer_created_event(3) # Tokyo timer
+
+    context_builder.add_orchestrator_completed_event()
+    context_builder.add_orchestrator_started_event()
+
+    ## fire timers
+    context_builder.add_timer_fired_event(id_=2, fire_at=fire_at) # Seattle timer
```

Comment:
nit: this `fire_at` will be the value from line 360, not from 353.

Ideally we would have two separate variables to keep this accurate, but it almost certainly will not interfere with any of our business logic.